components/controls/Pager.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div>
+    <BFormSelect
+      id="page-size"
+      v-model="perPage"
+      class="form-control"
+      :options="pageOptions"
+    />
+    <BPagination
+      v-model="currentPage"
+      :total-rows="recordCount"
+      :per-page="pageSize"
+      prev-text="Prev"
+      next-text="Next"
+      hide-goto-end-buttons
+      limit="4"
+    />
+  </div>
+</template>
+
+<script>
+import _ from 'lodash'
+import { BFormSelect, BPagination } from 'bootstrap-vue'
+
+const pageOptions = [
+  { value: 25, text: '25/page' },
+  { value: 50, text: '50/page' },
+  { value: 100, text: '100/page' },
+  { value: 250, text: '250/page' }
+]
+
+const defaultPageSize = pageOptions[0].value
+const defaultPage     = 1
+
+export default {
+  name      : 'ContactsList',
+  components: { BFormSelect, BPagination },
+  props     : {
+    pageSize:
+    {
+      type    : Number,
+      default : 25,
+      required: true,
+      validator (value){
+        return pageOptions.some(o => o.value == value)
+      }
+    },
+    page:
+    {
+      type    : Number,
+      required: true,
+      default : 1,
+      validator (value){
+        return value > 0
+      }
+    },
+    syncQueryString:
+    {
+      type    : Boolean,
+      required: true,
+      default : false
+    },
+    recordCount:
+    {
+      type    : Number,
+      required: true,
+      default : 0,
+      validator (value){
+        return value > -0.5
```

Comment:
What is this: ` value > -0.5 `????

Value must be greater or equal to 0
the test should be 

```suggestion
        return value >= 0
```

Reply:
sorry I read my previous comment. it was not wrriten correctly . you should hav read:
> Use value >=0. -1 look strange and **makes** -0.5 valid which is **wrong**.